### PR TITLE
Add an example config panel

### PR DESCRIPTION
**Pull request by lemniscate8**
Adding a configuration panel and some infrastructure to demonstrate use

### CODE REVIEW

**Comment by lemniscate8** [comment]
Looks like there's some problem with `<script src="https://cdn.jsdelivr.net/npm/less" ></script>` that's messing up styling and have yet to identity if it's due to recent changes.

File: {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/ProjectConfig.hpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+#include "emp/config/config.hpp"
+
+EMP_BUILD_CONFIG( {{ cookiecutter.project_name.replace(' ', '') }}Config,
+  GROUP(GLOBAL_SETTINGS, "Global settings"),
+  VALUE(SIZE, int, 300, "Size of example text area"),
+  VALUE(COLOR, std::string, "red", "Text color for example text area"),
+  GROUP(OTHER_PARAMETERS, "Miscellaneous settings"),
+  VALUE(LUNCH_ORDER, std::string, "ham on five", "What's for lunch today"),
+  VALUE(HOLD_MAYO, bool, true, "Whether or not to hold the mayo"),
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/ProjectConfig.hpp]: love it

File: {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/ProjectConfig.hpp [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#pragma once
+
+#include "emp/config/config.hpp"
+
+EMP_BUILD_CONFIG( {{ cookiecutter.project_name.replace(' ', '') }}Config,
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/ProjectConfig.hpp]: use cookiecutter.project_slug

Comment (mmore500) [reply {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/ProjectConfig.hpp]: ?

Comment (mmore500) [reply {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/ProjectConfig.hpp]: would need to change this everywhere else

Comment (mmore500) [reply {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/ProjectConfig.hpp]: oh, better: put an object `Config` inside of namespace `project_slug`

File: {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/example.hpp [hunk 3]
```diff
@@ -1,6 +1,35 @@
...
+  if(std::filesystem::exists("{{cookiecutter.project_slug}}.cfg")) {
+    std::cout << "Configuration read from {{cookiecutter.project_slug}}.cfg" << "\n";
+    config.Read("{{cookiecutter.project_slug}}.cfg");
+  }
+  am.UseCallbacks();
+  if (am.HasUnused())
+    std::exit(EXIT_FAILURE);
+}
+
+void SetupProjectConfig({{ cookiecutter.project_name.replace(' ', '') }}Config & config)  {
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/example.hpp]: rename so it's clear this is web entry point and one below is native entry point

File: {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/example.hpp [hunk 4]
```diff
@@ -1,6 +1,35 @@
...
+#include <filesystem>
+
+#include "emp/config/ArgManager.hpp"
+#include "emp/prefab/ConfigPanel.hpp"
+#include "emp/web/UrlParams.hpp"
+#include "emp/web/web.hpp"
+
+#include "{{cookiecutter.project_slug}}/ProjectConfig.hpp"
+
+void AddExistingConfig({{ cookiecutter.project_name.replace(' ', '') }}Config & config, emp::ArgManager & am) {
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/example.hpp]: I would put config stuff into its own header `config_setup.hpp` and make free functions `snake_case`

File: {{cookiecutter.project_slug}}/source/native.cpp [hunk 5]
```diff
@@ -5,17 +5,21 @@
...
+
+{{ cookiecutter.project_name.replace(' ', '') }}Config cfg;
 
 // This is the main function for the NATIVE version of {{cookiecutter.project_name}}.
 
 int main(int argc, char* argv[])
 {
-  emp::vector<std::string> args = emp::cl::args_to_strings(argc, argv);
-
   std::cout << "Hello, world!" << std::endl;
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/source/native.cpp]: we should switch std::endl for '\n'

File: {{cookiecutter.project_slug}}/source/native.cpp [hunk 6]
```diff
@@ -5,17 +5,21 @@
...
+  
+  // Set up a configuration panel for native application
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/source/native.cpp:19]: Print config settings

File: {{cookiecutter.project_slug}}/source/web.cpp [hunk 7]
```diff
@@ -4,17 +4,40 @@
...
 int main()
 {
   doc << "<h1>Hello, browser!</h1>";
+
+  // Set up a configuration panel for web application
+  SetupProjectConfig(cfg);
+  example_config_panel.Setup();
+  doc << example_config_panel.GetConfigPanelDiv();
+
+  // An example to show config usage
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/source/web.cpp]: Can you tell a little about what the example is supposed to do?

File: {{cookiecutter.project_slug}}/source/web.cpp [hunk 8]
```diff
@@ -4,17 +4,40 @@
...
 #include "{{cookiecutter.project_slug}}/example.hpp"
+#include "{{cookiecutter.project_slug}}/ProjectConfig.hpp"
 
 namespace UI = emp::web;
 
 UI::Document doc("emp_base");
 
+{{ cookiecutter.project_name.replace(' ', '') }}Config cfg;
+emp::prefab::ConfigPanel example_config_panel(cfg);
+UI::TextArea example_area;
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/source/web.cpp]: This can go into the scope of main

**Review by mmore500 — CHANGES_REQUESTED** [review]
A few minor stylistic/cosmetic changes, otherwise looking good!

File: {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/config_setup.hpp [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+#include "emp/config/ArgManager.hpp"
+#include "emp/prefab/ConfigPanel.hpp"
+#include "emp/web/UrlParams.hpp"
+#include "emp/web/web.hpp"
+
+#include "{{cookiecutter.project_slug}}/ExampleConfig.hpp"
+
+void use_existing_config_file({{ cookiecutter.project_slug.replace('-','_') }}::Config & config, emp::ArgManager & am) {
+  if(std::filesystem::exists("{{cookiecutter.project_slug}}.cfg")) {
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/config_setup.hpp:13]: space after if

File: {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/example.hpp [hunk 10]
```diff
@@ -3,6 +3,6 @@
 #include <iostream>
 
 bool example() {
-  std::cout << "example thing!" << std::endl;
+  std::cout << "example thing!" << "\n";
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/include/{{cookiecutter.project_slug}}/example.hpp:6]: use `'\n'`

File: {{cookiecutter.project_slug}}/source/native.cpp [hunk 11]
```diff
@@ -5,17 +5,22 @@
...
 int main(int argc, char* argv[])
-{
-  emp::vector<std::string> args = emp::cl::args_to_strings(argc, argv);
+{ 
+  // Set up a configuration panel for native application
+  setup_config_native(cfg, argc, argv);
+  cfg.Write(std::cout);
 
-  std::cout << "Hello, world!" << std::endl;
+  std::cout << "Hello, world!" << "\n";
```
Comment (mmore500) [inline {{cookiecutter.project_slug}}/source/native.cpp:23]: ibid

**Review by mmore500 — APPROVED** [review]
looks good. a few comments. We should be able to merge once we exorcise that segfault.

**Pull request merged**